pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -378,3 +446,7 @@ func (r *ReconcileUserSignup) updateStatusConditions(userSignup *toolchainv1alph
 	}
 	return r.client.Status().Update(context.TODO(), userSignup)
 }
+
+func (r *ReconcileUserSignup) updateStatusCompliantUsername(userSignup *toolchainv1alpha1.UserSignup) error {
```

Comment:
This function name is misleading. It's just updating the status. There is nothing compliant username specific. Not sure why do we need this function at all?